File: primatis-api/src/main/java/be/primatis/catalogue/dto/UpdateTitleStatusRequest.java
package be.primatis.catalogue.dto;

import be.primatis.catalogue.TitleStatus;
import jakarta.validation.constraints.NotNull;

/**
 * Contrat REST de transition {@code TitleStatus} (DEV-06.5,
 * {@code PATCH /api/v1/staff/titles/{titleId}/status}, {@code
 * CATALOGUE_MANAGE}). Seules {@code ACTIVE}/{@code WITHDRAWN} existent
 * (enum fermé) — aucune valeur supplémentaire n'est possible. Une valeur
 * textuelle hors enum échoue à la désérialisation Jackson elle-même (400
 * {@code MALFORMED_REQUEST_BODY}, infrastructure existante), avant que
 * {@code @NotNull} n'ait la moindre chance de s'appliquer.
 */
public record UpdateTitleStatusRequest(@NotNull TitleStatus status) {
}

File: primatis-web/src/app/catalogue/models/update-title-status-request.ts
import { TitleStatus } from './title-status';

/**
 * Voir `be.primatis.catalogue.dto.UpdateTitleStatusRequest` côté backend.
 * Champ nommé `status` (pas `titleStatus`) — nom de propriété backend exact.
 */
export interface UpdateTitleStatusRequest {
  status: TitleStatus;
}
